### DISCUSSION TITLE
Dealing with dump files...

### DISCUSSION THREAD

**Discussion by brendan-stephens** [psycopg/psycopg2#1674] [Q&A]
Does anyone have a good method for dealing with data loads from dump files?

For the most part, reading sql is fine:
` cs.copy_expert(sql=f.read(), file=f)`

Save that pg_dump will want to load data from stdin... 
```
COPY public.actor (actor_id, first_name, last_name, last_update) FROM stdin;
1       Penelope        Guiness 2013-05-26 14:47:57.62
2       Nick    Wahlberg        2013-05-26 14:47:57.62
3       Ed      Chase   2013-05-26 14:47:57.62
```

Then the subsequent data import will fail syntax.

**Comment by dvarrazzo** [comment]
`f.read()` doesn't seem to contain the query only, it seems to contain all the file.

Separate the query from the data.
